admin.php

```diff
@@ -206,22 +207,89 @@ function html() {
                 echo '<li>' . $form->toHTML() . '</li>';
                 echo '</ul>';
 
-                $form = new Doku_Form(array('class'=> 'sqliteplugin'));
+                /** @var $helper helper_plugin_sqlite */
+                $sqlite_db = plugin_load('helper', 'sqlite');
+                $sqlite_db->init('sqlite', DOKU_PLUGIN . 'sqlite/db/');
+
+                if($INPUT->str('action') == 'save') {
+                    $ok = true;
+                    if(empty($INPUT->str('sql'))) {
+                        msg($this->getLang('validation query_required'), -1);
+                        $ok = false;
+                    }
+                    if(empty($INPUT->str('name'))) {
+                        msg($this->getLang('validation query_name_required'), -1);
+                        $ok = false;
+                    }
+
+                    if($ok) {
+                        $sqlite_db->storeEntry('queries', array(
+                            'db' => $_REQUEST['db'],
+                            'name' => $INPUT->str('name'),
+                            'sql' => $INPUT->str('sql')
+                        ));
+                        msg($this->getLang('success query_saved'), 1);
+                    }
+                } elseif($INPUT->str('action') == 'delete') {
+                    $sqlite_db->query("DELETE FROM queries WHERE id=?;", $INPUT->int('query_id'));
+                    msg($this->getLang('success query_deleted'), 1);
+                }
```

Comment:
It seems the security token isn't checked here. It needs to be checked to prevent CSRF attacks.